### DISCUSSION TITLE
Three questions: flowSOM MST, diff expression, and volcano plot

### DISCUSSION THREAD

**Discussion by hrj21** [ImmuneDynamics/Spectre#30] [! News]
Hello! I have three questions but I wasn't sure if it was best to create separate posts.

1. Are able to plot and save an image of a minimal spanning tree from a flowSOM clustering? This is useful as an additional visualization step. I think something similar is available for Chronoclust clustering with `make.network.plot()`.
2. While it's nice that univariate statistical tests are available to test expression level differences between groups, are there plans to incorporate a regularization method (Lasso, elastic net etc) like in the diffcyt package?
3. I can see there is a function for constructing volcano plots, but I can't find documentation on how to use it. Do you have an example you could share?

Thanks for a wonderful package!
Hefin

**Comment by tomashhurst** [comment]
Hi @hrj21 , excellent questions! Thanks for reaching out.

1. We haven't included the minimal spanning tree plots by default, but this really just reflects a personal preference based on how we like to investigate the data. In the next version of Spectre we will be saving the MST structure, so the plots will be pretty simple. A middle-ground option which is fairly easy to implement is to create tSNE/UMAP plot of the clusters themselves -- so they wouldn't be connected by a spanning tree, but you would be able to visualise all the nodes and then plot them on tSNE/UMAP. You can always try using the [original FlowSOM package](https://github.com/SofieVG/FlowSOM) as well.

2. We don't have regularisation etc methods in this version of Spectre, as we essentially decided we would only provide simple statistical methods to get started. However we are working on implementing a few of these options. @ghar1821 would you like to comment?

3. That's my bad, I haven't made a meaningful set of instructions or a tutorial for it yet. However, we've just update the tutorials for summary data, heatmaps, and violin plots, and we'll have ones for running PCA and volcano plots done in the next week or so. Check in on the [tutorials page](https://immunedynamics.github.io/spectre/tutorials/) in a couple of days time, but I'll post an update here too when it's up to date.

Hope that's helpful!
